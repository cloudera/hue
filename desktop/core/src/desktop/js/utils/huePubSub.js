// Licensed to Cloudera, Inc. under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  Cloudera, Inc. licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// Based on original pub/sub implementation from http://davidwalsh.name/pubsub-javascript

const topics = {};
const hOP = topics.hasOwnProperty;

const huePubSub = {
  subscribe: function (topic, listener, app) {
    if (!hOP.call(topics, topic)) {
      topics[topic] = [];
    }

    const index =
      topics[topic].push({
        listener: listener,
        app: app,
        status: 'running'
      }) - 1;

    return {
      remove: function () {
        delete topics[topic][index];
      }
    };
  },
  removeAll: function (topic) {
    topics[topic] = [];
  },
  subscribeOnce: function (topic, listener, app) {
    const ephemeral = this.subscribe(
      topic,
      function () {
        listener.apply(listener, arguments);
        ephemeral.remove();
      },
      app
    );
  },
  publish: function (topic, info) {
    if (!hOP.call(topics, topic)) {
      return;
    }

    topics[topic].forEach(item => {
      if (item.status === 'running') {
        item.listener(info);
      }
    });
  },
  getTopics: function () {
    return topics;
  },
  pauseAppSubscribers: function (app) {
    if (app) {
      Object.keys(topics).forEach(topicName => {
        topics[topicName].forEach(topic => {
          if (
            typeof topic.app !== 'undefined' &&
            topic.app !== null &&
            (topic.app === app || topic.app.split('-')[0] === app)
          ) {
            topic.status = 'paused';
          }
        });
      });
    }
  },
  resumeAppSubscribers: function (app) {
    if (app) {
      Object.keys(topics).forEach(topicName => {
        topics[topicName].forEach(topic => {
          if (
            typeof topic.app !== 'undefined' &&
            topic.app !== null &&
            (topic.app === app || topic.app.split('-')[0] === app)
          ) {
            topic.status = 'running';
          }
        });
      });
    }
  },
  clearAppSubscribers: function (app) {
    if (app) {
      Object.keys(topics).forEach(topicName => {
        topics[topicName] = topics[topicName].filter(obj => {
          return obj.app !== app;
        });
      });
    }
  }
};

export default huePubSub;
