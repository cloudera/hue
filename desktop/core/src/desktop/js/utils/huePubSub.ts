// Licensed to Cloudera, Inc. under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  Cloudera, Inc. licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// Based on original pub/sub implementation from http://davidwalsh.name/pubsub-javascript

/* eslint-disable @typescript-eslint/no-explicit-any */

interface Listener<T> {
  (data: T): void;
}

enum PubSubState {
  RUNNING,
  PAUSED
}

const topics: {
  [topic: string]: {
    listener: Listener<any>;
    app?: string;
    state: PubSubState;
  }[];
} = {};

const hOP = topics.hasOwnProperty;

export interface HueSubscription {
  remove: () => void;
}

const subscribe = <T = any>(
  topic: string,
  listener: Listener<T>,
  app?: string
): HueSubscription => {
  if (!hOP.call(topics, topic)) {
    topics[topic] = [];
  }

  const index =
    topics[topic].push({
      listener: listener,
      app: app,
      state: PubSubState.RUNNING
    }) - 1;

  return {
    remove: () => {
      delete topics[topic][index];
    }
  };
};

const removeAll = (topic: string): void => {
  topics[topic] = [];
};

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
const publish = <T = any>(topic: string, info?: T): void => {
  if (!hOP.call(topics, topic)) {
    return;
  }

  topics[topic].forEach(item => {
    if (item.state === PubSubState.RUNNING) {
      item.listener(info);
    }
  });
};

const subscribeOnce = <T = any>(
  topic: string,
  listener: Listener<T>,
  app: string
): HueSubscription => {
  const ephemeral = subscribe(
    topic,
    arg => {
      listener(arg);
      ephemeral.remove();
    },
    app
  );
  return ephemeral;
};

const getTopics = (): typeof topics => topics;

const setStatusForApp = (state: PubSubState, app?: string): void => {
  if (app) {
    Object.keys(topics).forEach(topicName => {
      topics[topicName].forEach(topic => {
        if (topic.app && (topic.app === app || topic.app.split('-')[0] === app)) {
          topic.state = state;
        }
      });
    });
  }
};

const pauseAppSubscribers = (app?: string): void => {
  setStatusForApp(PubSubState.PAUSED, app);
};

const resumeAppSubscribers = (app?: string): void => {
  setStatusForApp(PubSubState.RUNNING, app);
};

const clearAppSubscribers = (app?: string): void => {
  if (app) {
    Object.keys(topics).forEach(topicName => {
      topics[topicName] = topics[topicName].filter(obj => {
        return obj.app !== app;
      });
    });
  }
};

export default {
  clearAppSubscribers,
  getTopics,
  pauseAppSubscribers,
  publish,
  removeAll,
  resumeAppSubscribers,
  subscribe,
  subscribeOnce
};
