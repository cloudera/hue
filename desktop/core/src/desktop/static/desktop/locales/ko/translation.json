{
  "Add": "추가",
  "aggregate": "집계",
  "Aggregate": "집계",
  "alias": "별칭",
  "All": "모두",
  "and": "및",
  "Apps": "앱",
  "Assist": "지원",
  "Assistant": "도우미",
  "Back": "뒤로",
  "Browsers": "브라우저",
  "Bundle": "Bundle",
  "Cancel": "취소",
  "Change": "변경",
  "Clipboard": "클립보드",
  "Close": "닫기",
  "Clusters": "클러스터",
  "Columns": "열",
  "condition": "조건",
  "Create": "생성",
  "Created": "생성됨",
  "CSV": "CSV",
  "cte": "cte",
  "CTEs": "CTE",
  "Dashboard": "대시보드",
  "database": "데이터베이스",
  "Databases": "데이터베이스",
  "Date": "날짜",
  "default": "기본값:",
  "Delete": "삭제",
  "Description": "설명",
  "Details": "세부 정보",
  "Directory": "디렉토리",
  "Documentation": "문서",
  "Documents": "문서",
  "Editor": "편집기",
  "Excel": "Excel",
  "Export": "내보내기",
  "Failed": "실패함",
  "Field": "필드",
  "File": "파일",
  "Files": "파일",
  "filter": "필터",
  "Filter": "필터",
  "Format": "형식",
  "Functions": "함수",
  "Group": "그룹",
  "HBase": "HBase",
  "Help": "도움말",
  "Hive": "Hive",
  "Identifiers": "식별자",
  "Impala": "Impala",
  "Import": "가져오기",
  "Indexes": "인덱스",
  "Insert": "커서 위치에 ",
  "Jobs": "Job",
  "join": "가입",
  "Key": "키",
  "keyword": "키워드",
  "Keywords": "키워드",
  "Metrics": "메트릭",
  "Modify": "수정",
  "More": "더",
  "Name": "이름",
  "Next": "다음",
  "No": "아니요",
  "Notebook": "노트북",
  "of": "/",
  "Open": "열기",
  "Options": "옵션",
  "Order": "순서",
  "other": "기타",
  "Output": "출력",
  "Owner": "소유자",
  "Partitions": "파티션",
  "Password": "암호",
  "Permissions": "권한",
  "Popular": "인기",
  "Popularity": "인기도",
  "Project": "프로젝트",
  "Queries": "쿼리",
  "Read": "읽기",
  "Refresh": "새로 고침",
  "Remove": "제거",
  "Report": "보고서",
  "Running": "실행 중",
  "Sample": "샘플",
  "sample": "샘플",
  "Samples": "샘플",
  "Schedule": "예약",
  "Search": "검색",    
  "Settings": "설정",
  "Share": "공유",
  "Sharing": "공유",
  "Size": "크기",
  "Sources": "소스",
  "SQL": "SQL",
  "Start": "시작",
  "Statement": "명령문",
  "Stats": "상태",
  "Stopped": "중지됨",
  "table": "테이블",
  "Table": "테이블",
  "Tables": "테이블",
  "Terms": "용어",
  "Type": "유형",
  "UDFs": "UDF",
  "Undo": "실행 취소",
  "Username": "사용자 이름",
  "Value": "값",
  "Values": "값",
  "variable": "변수",
  "Variables": "변수",
  "View": "보기",
  "view": "보기",
  "Views": "보기",
  "virtual": "가상",
  "Workflow": "Workflow",
  "Yes": "예"
}