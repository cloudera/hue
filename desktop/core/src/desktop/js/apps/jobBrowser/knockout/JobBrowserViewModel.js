// Licensed to Cloudera, Inc. under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  Cloudera, Inc. licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import $ from 'jquery';
import * as ko from 'knockout';

import { getLastKnownConfig } from 'config/hueConfig';
import { GLOBAL_ERROR_TOPIC } from 'reactComponents/AlertComponent/events';
import huePubSub from 'utils/huePubSub';
import I18n from 'utils/i18n';
import { withLocalStorage } from 'utils/storageUtils';
import changeURL from 'utils/url/changeURL';

import Job from './Job';
import Jobs from './Jobs';

export default class JobBrowserViewModel {
  constructor(isMiniJobBrowser) {
    this.assistAvailable = ko.observable(true);
    this.isLeftPanelVisible = ko.observable();
    withLocalStorage('assist.assist_panel_visible', this.isLeftPanelVisible, true);
    this.isLeftPanelVisible.subscribe(() => {
      huePubSub.publish('assist.forceRender');
    });

    this.appConfig = ko.observable();
    this.clusterType = ko.observable();
    this.isMini = ko.observable(isMiniJobBrowser);

    this.cluster = ko.observable();
    this.compute = ko.observable();
    this.compute.subscribe(() => {
      if (this.interface()) {
        this.jobs.fetchJobs();
      }
    });

    this.availableInterfaces = ko.pureComputed(() => {
      const isDialectEnabled = dialect =>
        this.appConfig()?.editor?.interpreter_names?.indexOf(dialect) >= 0;

      const historyInterfaceCondition = () => window.ENABLE_HISTORY_V2;

      const jobsInterfaceCondition = () =>
        !getLastKnownConfig().has_computes &&
        this.appConfig()?.browser?.interpreter_names.indexOf('yarn') !== -1 &&
        (!this.cluster() || this.cluster().type.indexOf('altus') === -1);

      const dataEngInterfaceCondition = () => this.cluster()?.type === 'altus-de';

      const enginesInterfaceCondition = () => this.cluster()?.type === 'altus-engines';

      const dataWarehouseInterfaceCondition = () => this.cluster()?.type === 'altus-dw';

      const dataWarehouse2InterfaceCondition = () => this.cluster()?.type === 'altus-dw2';

      const schedulerInterfaceCondition = () =>
        window.USER_HAS_OOZIE_ACCESS &&
        (!this.cluster() || this.cluster().type.indexOf('altus') === -1) &&
        this.appConfig()?.scheduler;

      const schedulerExtraInterfaceCondition = () =>
        !this.isMini() && schedulerInterfaceCondition();

      const schedulerBeatInterfaceCondition = () =>
        this.appConfig()?.scheduler?.interpreter_names.indexOf('celery-beat') !== -1;

      const livyInterfaceCondition = () =>
        !this.isMini() &&
        this.appConfig()?.editor &&
        (this.appConfig().editor.interpreter_names.indexOf('pyspark') !== -1 ||
          this.appConfig().editor.interpreter_names.indexOf('sparksql') !== -1);

      const queryInterfaceCondition = () =>
        window.ENABLE_QUERY_BROWSER &&
        !getLastKnownConfig().has_computes &&
        this.appConfig()?.editor.interpreter_names.indexOf('impala') !== -1 &&
        (!this.cluster() || this.cluster().type.indexOf('altus') === -1);

      const queryHiveInterfaceCondition = () => {
        return window.ENABLE_HIVE_QUERY_BROWSER && !getLastKnownConfig().has_computes;
      };

      const scheduleHiveInterfaceCondition = () => window.ENABLE_QUERY_SCHEDULING;

      const hiveQueriesInterfaceCondition = () =>
        window.ENABLE_QUERY_STORE && isDialectEnabled('hive');
      const impalaQueriesInterfaceCondition = () =>
        window.ENABLE_QUERY_STORE && isDialectEnabled('impala');

      const interfaces = [
        { interface: 'jobs', label: I18n('Jobs'), condition: jobsInterfaceCondition },
        { interface: 'dataeng-jobs', label: I18n('Jobs'), condition: dataEngInterfaceCondition },
        {
          interface: 'dataeng-clusters',
          label: I18n('Clusters'),
          condition: dataEngInterfaceCondition
        },
        {
          interface: 'dataware-clusters',
          label: I18n('Clusters'),
          condition: dataWarehouseInterfaceCondition
        },
        {
          interface: 'dataware2-clusters',
          label: I18n('Warehouses'),
          condition: dataWarehouse2InterfaceCondition
        },
        { interface: 'engines', label: '', condition: enginesInterfaceCondition },
        { interface: 'queries-impala', label: 'Impala', condition: queryInterfaceCondition },
        { interface: 'queries-hive', label: 'Hive', condition: queryHiveInterfaceCondition },
        {
          interface: 'schedule-hive',
          label: I18n('Hive Schedules'),
          condition: scheduleHiveInterfaceCondition
        },
        {
          interface: 'celery-beat',
          label: I18n('Scheduled Tasks'),
          condition: schedulerBeatInterfaceCondition
        },
        { interface: 'history', label: I18n('History'), condition: historyInterfaceCondition },
        {
          interface: 'workflows',
          label: I18n('Workflows'),
          condition: schedulerInterfaceCondition
        },
        {
          interface: 'schedules',
          label: I18n('Schedules'),
          condition: schedulerInterfaceCondition
        },
        {
          interface: 'bundles',
          label: I18n('Bundles'),
          condition: schedulerExtraInterfaceCondition
        },
        { interface: 'slas', label: I18n('SLAs'), condition: schedulerExtraInterfaceCondition },
        { interface: 'livy-sessions', label: 'Livy', condition: livyInterfaceCondition },
        {
          interface: 'hive-queries',
          label: I18n('Hive Queries'),
          condition: hiveQueriesInterfaceCondition
        },
        {
          interface: 'impala-queries',
          label: I18n('Impala Queries'),
          condition: impalaQueriesInterfaceCondition
        }
      ];

      return interfaces.filter(i => i.condition());
    });

    this.availableInterfaces.subscribe(newInterfaces => {
      if (
        this.interface() &&
        !newInterfaces.some(newInterface => newInterface.interface === this.interface())
      ) {
        this.selectInterface(newInterfaces[0]);
      }
    });

    this.slasLoadedOnce = false;
    this.slasLoading = ko.observable(true);

    this.oozieInfoLoadedOnce = false;
    this.oozieInfoLoading = ko.observable(true);

    this.interface = ko.observable();

    this.jobs = new Jobs(this);
    this.job = ko.observable();

    this.updateJobTimeout = -1;
    this.updateJobsTimeout = -1;
    this.jobUpdateCounter = 0;
    this.exponentialFactor = 1;

    this.job.subscribe(val => {
      this.monitorJob(val);
    });

    this.breadcrumbs = ko.observableArray([]);

    this.resetBreadcrumbs();
  }

  contextId(id) {
    if (this.isMini()) {
      return id + '-mini';
    }
    return id;
  }

  showTab(anchor) {
    $(`a[href='${this.contextId(anchor)}']`).tab('show');
  }

  loadSlaPage() {
    if (!this.slasLoadedOnce) {
      $.ajax({
        url: '/oozie/list_oozie_sla/?is_embeddable=true',
        beforeSend: xhr => {
          xhr.setRequestHeader('X-Requested-With', 'Hue');
        },
        dataType: 'html',
        success: response => {
          this.slasLoading(false);
          $('#slas').html(response);
        }
      });
    }
  }

  loadOozieInfoPage() {
    if (!this.oozieInfoLoadedOnce) {
      this.oozieInfoLoadedOnce = true;
      this.oozieInfoLoading(true);
      $.ajax({
        url: '/oozie/list_oozie_info/?is_embeddable=true',
        beforeSend: xhr => {
          xhr.setRequestHeader('X-Requested-With', 'Hue');
        },
        dataType: 'html'
      })
        .done(response => {
          $('#oozieInfo').html(response);
        })
        .fail(err => {
          this.oozieInfoLoadedOnce = false;
          this.selectInterface('schedules');
          huePubSub.publish(GLOBAL_ERROR_TOPIC, { message: I18n('Failed loading Oozie Info.') });
        })
        .always(() => {
          this.oozieInfoLoading(false);
        });
    }
  }

  isValidInterface(name) {
    const flatAvailableInterfaces = this.availableInterfaces().map(i => i.interface);
    if (flatAvailableInterfaces.indexOf(name) !== -1 || name === 'oozie-info') {
      return name;
    } else {
      return flatAvailableInterfaces[0];
    }
  }

  selectInterface(selectedInterface) {
    const validSelectedInterface = this.isValidInterface(selectedInterface);
    this.interface(validSelectedInterface);
    this.resetBreadcrumbs();

    if (!this.isMini()) {
      huePubSub.publish('graph.stop.refresh.view');
      if (window.location.hash !== '#!' + validSelectedInterface) {
        changeURL('#!' + validSelectedInterface);
      }
    }
    this.jobs.selectedJobs([]);
    this.job(null);

    if (validSelectedInterface === 'slas' && !this.isMini()) {
      this.loadSlaPage();
    } else if (validSelectedInterface === 'oozie-info' && !this.isMini()) {
      this.loadOozieInfoPage();
    } else if (
      validSelectedInterface !== 'hive-queries' ||
      validSelectedInterface !== 'impala-queries'
    ) {
      this.jobs.fetchJobs();
    }
  }

  onClusterSelect() {
    let interfaceToSet = this.interface();
    if (
      !this.availableInterfaces().some(
        availableInterface => availableInterface.interface === interfaceToSet
      )
    ) {
      interfaceToSet = this.availableInterfaces()[0].interface;
    }
    this.selectInterface(interfaceToSet);
  }

  monitorJob(job) {
    window.clearTimeout(this.updateJobTimeout);
    window.clearTimeout(this.updateJobsTimeout);
    this.jobUpdateCounter = 0;
    this.exponentialFactor = 1;
    if (
      this.interface() &&
      this.interface() !== 'slas' &&
      this.interface() !== 'oozie-info' &&
      this.interface !== 'hive-queries' &&
      this.interface !== 'impala-queries'
    ) {
      if (job) {
        if (job.apiStatus() === 'RUNNING') {
          const _updateJob = () => {
            this.jobUpdateCounter++;
            const updateLogs = this.jobUpdateCounter % this.exponentialFactor === 0;
            if (updateLogs && this.exponentialFactor < 50) {
              this.exponentialFactor *= 2;
            }
            const def = job.updateJob(updateLogs);
            if (def) {
              def.done(() => {
                this.updateJobTimeout = setTimeout(
                  _updateJob,
                  window.JB_SINGLE_CHECK_INTERVAL_IN_MILLIS
                );
              });
            }
          };
          this.updateJobTimeout = setTimeout(_updateJob, window.JB_SINGLE_CHECK_INTERVAL_IN_MILLIS);
        }
      } else {
        const _updateJobs = () => {
          const updateRequest = this.jobs.updateJobs();
          if (updateRequest?.done) {
            updateRequest.done(() => {
              setTimeout(_updateJobs, window.JB_MULTI_CHECK_INTERVAL_IN_MILLIS);
            });
          }
        };
        this.updateJobsTimeout = setTimeout(_updateJobs, window.JB_MULTI_CHECK_INTERVAL_IN_MILLIS);
      }
    }
  }

  resetBreadcrumbs(extraCrumbs) {
    let crumbs = [{ id: '', name: this.interface(), type: this.interface() }];
    if (extraCrumbs) {
      crumbs = crumbs.concat(extraCrumbs);
    }
    this.breadcrumbs(crumbs);
  }

  getHDFSPath(path) {
    if (path.startsWith('hdfs://')) {
      const bits = path.substr(7).split('/');
      bits.shift();
      return '/' + bits.join('/');
    }
    return path;
  }

  formatProgress(progress) {
    if (typeof progress === 'function') {
      progress = progress();
    }
    if (!isNaN(progress)) {
      return Math.round(progress * 100) / 100 + '%';
    }
    return progress;
  }

  load() {
    let h = window.location.hash;
    if (!this.isMini()) {
      huePubSub.publish('graph.stop.refresh.view');
    }

    h = h.indexOf('#!') === 0 ? h.substr(2) : '';
    switch (h) {
      case '':
        h = 'jobs';
      case 'slas':
      case 'oozie-info':
      case 'jobs':
      case 'queries-impala':
      case 'queries-hive':
      case 'celery-beat':
      case 'schedule-hive':
      case 'history':
      case 'workflows':
      case 'schedules':
      case 'bundles':
      case 'dataeng-clusters':
      case 'dataware-clusters':
      case 'dataware2-clusters':
      case 'engines':
      case 'dataeng-jobs':
      case 'livy-sessions':
      case 'hive-queries':
      case 'impala-queries':
        this.selectInterface(h);
        break;
      default:
        if (h.indexOf('id=') === 0 && !this.isMini()) {
          new Job(this, { id: h.substr(3) }).fetchJob();
        } else {
          this.selectInterface('reset');
        }
    }
  }
}
