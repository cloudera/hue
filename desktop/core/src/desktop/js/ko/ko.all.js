// Licensed to Cloudera, Inc. under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  Cloudera, Inc. licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import * as ko from 'knockout';
import 'ko/ko.init';
import komapping from 'knockout.mapping';
import 'knockout-switch-case';
import 'knockout-sortable';
import 'knockout.validation';

import 'ext/ko.editable.custom';
import 'ext/ko.selectize.custom';

import 'ko/bindings/charts/mapchart/ko.mapChart';
import 'ko/bindings/charts/ko.barChart';
import 'ko/bindings/charts/ko.leafletMapChart';
import 'ko/bindings/charts/ko.lineChart';
import 'ko/bindings/charts/ko.partitionChart';
import 'ko/bindings/charts/ko.pieChart';
import 'ko/bindings/charts/ko.scatterChart';
import 'ko/bindings/charts/ko.timelineChart';

import 'ko/bindings/ko.vue.js';

import 'ko/bindings/ace/ko.aceEditor';
import 'ko/bindings/ace/ko.aceResizer';
import 'ko/bindings/ko.appAwareTemplateContextMenu';
import 'ko/bindings/ko.assistFileDraggable';
import 'ko/bindings/ko.assistFileDroppable';
import 'ko/bindings/ko.assistVerticalResizer';
import 'ko/bindings/ko.attachViewModelToElementData';
import 'ko/bindings/ko.augmentHtml';
import 'ko/bindings/ko.autocomplete';
import 'ko/bindings/ko.autogrowInput';
import 'ko/bindings/ko.blurHide';
import 'ko/bindings/ko.bytesize';
import 'ko/bindings/ko.chosen';
import 'ko/bindings/ko.clearable';
import 'ko/bindings/ko.clickForAceFocus';
import 'ko/bindings/ko.clickToCopy';
import 'ko/bindings/ko.clipboard';
import 'ko/bindings/ko.codemirror';
import 'ko/bindings/ko.contextMenu';
import 'ko/bindings/ko.contextSubMenu';
import 'ko/bindings/ko.datalist';
import 'ko/bindings/ko.datepicker';
import 'ko/bindings/ko.dateRangePicker';
import 'ko/bindings/ko.dblClick';
import 'ko/bindings/ko.delayedOverflow';
import 'ko/bindings/ko.dockable';
import 'ko/bindings/ko.documentChooser';
import 'ko/bindings/ko.documentContextPopover';
import 'ko/bindings/ko.draggableText';
import 'ko/bindings/ko.dropdown';
import 'ko/bindings/ko.dropzone';
import 'ko/bindings/ko.duration';
import 'ko/bindings/ko.ellipsis';
import 'ko/bindings/ko.fadeVisible';
import 'ko/bindings/ko.fetchMore';
import 'ko/bindings/ko.fileChooser';
import 'ko/bindings/ko.foreachVisible';
import 'ko/bindings/ko.fresherEditor';
import 'ko/bindings/ko.hdfsAutocomplete';
import 'ko/bindings/ko.hdfsTree';
import 'ko/bindings/ko.highlight';
import 'ko/bindings/ko.hiveChooser';
import 'ko/bindings/ko.html';
import 'ko/bindings/ko.hueach';
import 'ko/bindings/ko.hueCheckAll';
import 'ko/bindings/ko.hueCheckbox';
import 'ko/bindings/ko.hueChecked';
import 'ko/bindings/ko.hueLink';
import 'ko/bindings/ko.hueSpinner';
import 'ko/bindings/ko.impalaDagre';
import 'ko/bindings/ko.jHueRowSelector';
import 'ko/bindings/ko.logResizer';
import 'ko/bindings/ko.logScroller';
import 'ko/bindings/ko.medium';
import 'ko/bindings/ko.moment';
import 'ko/bindings/ko.momentFromNow';
import 'ko/bindings/ko.multiCheck';
import 'ko/bindings/ko.multiCheckForeachVisible';
import 'ko/bindings/ko.multiClick';
import 'ko/bindings/ko.multiLineEllipsis';
import 'ko/bindings/ko.numberFormat';
import 'ko/bindings/ko.numericTextInput';
import 'ko/bindings/ko.onClickOutside';
import 'ko/bindings/ko.oneClickSelect';
import 'ko/bindings/ko.parseArguments';
import 'ko/bindings/ko.publish';
import 'ko/bindings/ko.reactWrapper';
import 'ko/bindings/ko.readOnlyAce';
import 'ko/bindings/ko.resizable';
import 'ko/bindings/ko.select2';
import 'ko/bindings/ko.simplesize';
import 'ko/bindings/ko.slider';
import 'ko/bindings/ko.slideVisible';
import 'ko/bindings/ko.solrChooser';
import 'ko/bindings/ko.spinEdit';
import 'ko/bindings/ko.splitDraggable';
import 'ko/bindings/ko.splitFlexDraggable';
import 'ko/bindings/ko.sqlContextPopover';
import 'ko/bindings/ko.stickVertical';
import 'ko/bindings/ko.storageContextPopover';
import 'ko/bindings/ko.stretchDown';
import 'ko/bindings/ko.tagEditor';
import 'ko/bindings/ko.tagsNotAllowed';
import 'ko/bindings/ko.templateContextMenu';
import 'ko/bindings/ko.templatePopover';
import 'ko/bindings/ko.textSqueezer';
import 'ko/bindings/ko.toggle';
import 'ko/bindings/ko.toggleOverflow';
import 'ko/bindings/ko.tooltip';
import 'ko/bindings/ko.truncatedText';
import 'ko/bindings/ko.typeahead';
import 'ko/bindings/ko.verticalSlide';
import 'ko/bindings/ko.visibleOnHover';

import 'ko/components/appSwitcher/ko.appSwitcher';
import 'ko/components/assist/ko.assistDashboardPanel';
import 'ko/components/assist/ko.assistDbPanel';
import 'ko/components/assist/ko.assistDocumentsPanel';
import 'ko/components/assist/ko.assistEditorContextPanel';
import 'ko/components/assist/ko.assistFunctionsPanel';
import 'ko/components/assist/ko.assistGitPanel';
import 'ko/components/assist/ko.assistHBasePanel';
import 'ko/components/assist/ko.assistLangRefPanel';
import 'ko/components/assist/ko.assistPanel';
import 'ko/components/assist/ko.assistSchedulePanel';
import 'ko/components/assist/ko.assistStoragePanel';
import 'ko/components/assist/ko.rightAssistPanel';
import 'ko/components/contextPopover/ko.contextPopover';
import 'ko/components/contextPopover/ko.quickQueryContext';
import 'ko/components/contextPopover/ko.popoverOnHover';
import 'ko/components/doc/ko.shareDocModal';
import 'ko/components/simpleAceEditor/ko.simpleAceEditor';

import 'ko/components/ko.catalogEntriesList';
import 'ko/components/ko.contextSelector';
import 'ko/components/ko.createDirectoryModal';
import 'ko/components/ko.deleteDocModal';
import 'ko/components/ko.dropDown';
import 'ko/components/ko.editorDroppableMenu';
import 'ko/components/ko.executionAnalysis';
import 'ko/components/ko.favoriteApp';
import 'ko/components/ko.fieldSamples';
import 'ko/components/ko.globalSearch';
import 'ko/components/ko.historyPanel';
import 'ko/components/ko.importDocumentsModal';
import 'ko/components/ko.inlineAutocomplete';
import 'ko/components/ko.jobBrowserLinks';
import 'ko/components/ko.navProperties';
import 'ko/components/ko.navTags';
import 'ko/components/ko.pollingCatalogEntriesList';
import 'ko/components/ko.quickQueryAction';
import 'ko/components/ko.sentryPrivileges';
import 'ko/components/ko.sessionAuthModal';
import 'ko/components/ko.sessionPanel';
import 'ko/components/ko.shareGistModal';
import 'ko/components/ko.sqlColumnsTable';

// TODO: Move to about app when it has it's own webpack entry
import 'apps/about/components/ko.hueConfigTree';
import 'apps/about/components/ko.connectorsConfig';

import 'ko/extenders/ko.maxLength';
import 'ko/extenders/ko.numeric';
import 'ko/extenders/ko.toJson';

import 'ko/observables/ko.observableArrayDefault';
import 'ko/observables/ko.observableDefault';

window.ko = ko;
window.ko.mapping = komapping;

export default ko;
