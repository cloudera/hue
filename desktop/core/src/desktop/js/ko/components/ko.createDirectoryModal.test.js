// Licensed to Cloudera, Inc. under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  Cloudera, Inc. licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import * as ko from 'knockout';
import { SHOWN_EVENT, SHOW_EVENT } from './ko.createDirectoryModal';
import huePubSub from 'utils/huePubSub';

describe('ko.createDirectoryModal.js', () => {
  it('should render component', async () => {
    huePubSub.publish(SHOW_EVENT, {
      activeEntry: ko.observable({}),
      isTrash: ko.observable(false),
      isTrashed: ko.observable(false),
      canModify: ko.observable(true)
    });

    await new Promise(resolve => {
      huePubSub.subscribeOnce(SHOWN_EVENT, resolve);
    });

    expect(window.document.documentElement.outerHTML).toMatchSnapshot();
  });
});
