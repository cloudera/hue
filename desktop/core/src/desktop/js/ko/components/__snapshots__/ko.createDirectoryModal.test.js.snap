// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ko.createDirectoryModal.js should render component 1`] = `
"<html><head></head><body class=\\"modal-open\\"><div id=\\"createDirectoryModal\\" data-bind=\\"descendantsComplete: descendantsComplete, component: { name: 'create-directory', params: params }\\" data-keyboard=\\"true\\" class=\\"modal hide fade in\\" tabindex=\\"-1\\" aria-hidden=\\"false\\">
  <!-- ko with: activeEntry -->
  <form class=\\"form-horizontal\\">
    <div class=\\"modal-header\\">
      <button type=\\"button\\" class=\\"close\\" data-dismiss=\\"modal\\" aria-label=\\"Close\\"><span aria-hidden=\\"true\\">Ã—</span></button>
      <h2 class=\\"modal-title\\">Create Directory</h2>
    </div>
    <div class=\\"modal-body \\">
      <input id=\\"newDirectoryName\\" class=\\"input large-as-modal\\" type=\\"text\\" placeholder=\\"Directory name\\">
    </div>
    <div class=\\"modal-footer\\">
      <input type=\\"button\\" class=\\"btn\\" data-dismiss=\\"modal\\" data-bind=\\"click: function () { $('#newDirectoryName').val(null) }\\" value=\\"Cancel\\">
      <input type=\\"submit\\" class=\\"btn btn-primary disable-feedback\\" value=\\"Create\\" data-bind=\\"click: function () { if ($('#newDirectoryName').val()) { $data.createDirectory($('#newDirectoryName').val()); $('#createDirectoryModal').modal('hide'); } }\\">
    </div>
  </form>
  <!-- /ko -->
</div><div class=\\"modal-backdrop fade in\\"></div></body></html>"
`;
